### PR TITLE
Fix "reference to free variable" warning: `git-timemachine-revision`

### PR DESCRIPTION
**Pull request by kaushalmodi**

### CODE REVIEW

File: git-link.el [hunk 1]
```diff
@@ -90,6 +90,9 @@
 (require 'thingatpt)
 (require 'url-util)
 
+(eval-when-compile
+  (defvar git-timemachine-revision))    ;silence reference to free variable warning
```
Comment (sshaw) [inline git-link.el]: I think`defvar` enough, no?

Comment (kaushalmodi) [inline git-link.el]: That leads to minor optimization as git-link does not need to evaluate that each time it is loaded. All we need is to tell the compiler that variable is an intended global declaration. https://www.gnu.org/software/emacs/manual/html_node/elisp/Eval-During-Compile.html

Comment (sshaw) [inline git-link.el]: > That leads to minor optimization...

Minor or micro 😸?

**Comment by kaushalmodi** [comment]
Thanks :)

**Pull request merged**